blog/content/post/2021-10-10-optiplex-dasharo.md

```diff
@@ -0,0 +1,173 @@
+has expressed his deep dissatisfaction how Intel phased off the ECC support on
+consumer chipsets in [this article](https://www.extremetech.com/computing/318832-linus-tovalds-blames-intel-for-killing-ecc-ram-in-consumer-systems)
+Maybe if we know a little bit more about how the autonegotiation works or if ME
+firmware may force the ECC support, it will be possible to override the current
+state.
+
+## NVMe support
+
+Dasharo firmware enables support for booting from PCIe NVMe drives unlike the
+original Dell firmware. Back in 2013 the first NVMe drives have actually been
+manufactured (Samsung XS1715) so the technology was not so popular when the
+machine is launched. However even after 5 years of [BIOS update
+releases](https://www.dell.com/support/home/en-us/drivers/driversdetails?driverid=90dd2&driverid=90dd2&lwp=rt)
+this lovely machine did not live to see NVMe support. These machines have a
+PCIe x4 slot which is perfect for a NVMe disk, just need a few bucks to buy an
+adapter like [this one](https://www.aliexpress.com/item/1005003114626058.html).
+With Dasharo firmware it is possible to boot from such NVMe disk without
+issues.
+
+## Intel TXT support and TrenchBoot
+
+Intel Trusted Execution Technology is a feature of Intel CPUs and chipsets to
+perform trusted measurement of the operation system software defined in Trusted
+Computing Group [D-RTM architecture specification](https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf).
+Although Dell firmware supports TXT and it is nothing new for Dasharo to
+support it as well, we haven't yet been able to successfully launch TXT with
+open source firmware on the Dell OptiPlex 7010 / 9010. But we intend to resolve
+this issue. Moreover we aim to support [TrenchBoot](https://trenchboot.org/)
+on this machine. A very good use case of Intel TXT was presented by Qubes OS
+[Anti Evil Maid (AEM)](https://github.com/QubesOS/qubes-antievilmaid/). By the
+way we have been working on enabling AMD CPUs and TPM 2.0 as well in AEM.
+In an ideal case Dell OptiPlex will come with TXT and TrenchBoot support and
+use AEM to securely measure Qubes OS and perform a remote attestation using
+[Fobnail](https://fobnail.3mdeb.com/). To protect the user even further we also
+plan to integrate [Intel STM](https://software.intel.com/content/www/us/en/develop/articles/smi-transfer-monitor-stm.html)
+to avoid System Management Mode attacks to which Intel TXT is vulnerable.
+
+## Open Security Training 2
+
+All these possibilities, open source firmware and security features makes the
+Dell OptiPlex 7010 / 9010 a first class citizen secure workstation that has been
+recognized by [Xeno Kovah](https://twitter.com/XenoKovah), a former employee of
+Apple who worked in Apple's security department. Dell OptiPlex 7010 / 9010 is the
+reference machine used in the [Open Security Training 2](https://ost2.fyi/) - a
+freely available firmware and security training, second edition.
```

Comment:
Well, to say it's "a first class citizen secure workstation that has been recognized by" me, would imply that I had done an audit of the security and found no problems. Since I haven't, perhaps a better phrasing would be:

"
All these possibilities, open source firmware and security features makes the
Dell OptiPlex 7010 / 9010 a first class citizen secure workstation."

That then sounds more like your assertion, rather than mine. Followed by
"
It has been recognized by [Xeno Kovah](https://twitter.com/XenoKovah), a former Apple firmware security architect, as an excellent training platform for teaching Intel firmware security and coreboot courses. The Dell OptiPlex 7010 / 9010 is used as a reference machine in the [OpenSecurityTraining2](https://ost2.fyi/) "Architecture 4001: x86-64 Reset Vector Firmware" class (which is currently in private beta testing).
"

(We can update this with a link to the class later when it's out of beta if so desired.)

"
And that class then leads into the ["Architecture 4031: x86-64 Reset Vector: coreboot"](https://p.ost2.fyi/courses/course-v1:OpenSecurityTraining2+Arch4031_x86-64_RV_coreboot+2021_v1/about) class, and future coreboot classes which are under development, such as "Architecture 4032: coreboot Hardware Hands-On".
"

Reply:
Done. @XenoKovah please check if the form suits you now.